chainer/functions/normalization/batch_normalization.py

```diff
@@ -15,6 +17,227 @@
 import chainerx
 
 
+class _BatchNormalizationBackend(six.with_metaclass(ABCMeta)):
+
+    @abstractmethod
+    def forward(self, axis, gamma, x, xp, expander,
+                beta, eps, decay, running_mean, running_var):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def backward(self, axis, gamma, gy, x, xp,
+                 expander, mean, inv_std, eps, var, mode):
+        raise NotImplementedError()
+
+
+class _GeneralBatchNormalizationBackend(_BatchNormalizationBackend):
+
+    def __init__(self, mode=None):
```

Comment:
I think `_BNMode` can now be integrated into backend selectors.
How do you think?

Reply:
You mean putting fields in `_BNMode` to corresponding `backend`?
Since `_BNMode` is also used in `FixedBatchNormalizationGrad`, removing it means we also need to refactor `FixedBatchNormalizationGrad`, which is not included in this patch.
Do you think we should do it in this PR? or another PR.

Reply:
>You mean putting fields in `_BNMode` to corresponding `backend`?

No, I meant "selectors", not "backends". The role of `_BNMode` is overlapping with selectors.

OK, I understand it's not easy to simply eliminate `_BNMode`. But I don't think it doesn't have to be passed to backends. Their only usage is `is_for_conv2d` and `get_cudnn_mode` in `_CudnnBatchNormalizationBackend`. How about passing those two values to `_CudnnBatchNormalizationBackend.__init__`, instead of passing the `mode`?

Reply:
Addressed in 918119f